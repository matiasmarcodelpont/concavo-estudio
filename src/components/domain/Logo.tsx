import Image from 'next/image'
import React from 'react'

interface LogoProps {
  className?: string
}

const Logo = ({ className }: LogoProps) => {
  return <Image width={200} height={200 / 3} src='/logos/concavo_black.png' alt='Logo CÃ³ncavo' className={className} />
}

export default Logo
