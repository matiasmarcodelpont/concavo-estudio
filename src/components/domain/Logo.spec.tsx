import '@testing-library/jest-dom'
import { render } from '@testing-library/react'

import Logo from './Logo'

jest.mock('next/image', () => ({
  __esModule: true,
  default: (
    props: JSX.IntrinsicAttributes &
      React.ClassAttributes<HTMLImageElement> &
      React.ImgHTMLAttributes<HTMLImageElement>,
    // eslint-disable-next-line @next/next/no-img-element, jsx-a11y/alt-text
  ) => <img {...props} />,
}))

describe('Logo', () => {
  it('matches snapshot', () => {
    const { getByRole } = render(<Logo />)
    const component = getByRole('img')

    expect(component).toMatchSnapshot()
  })

  it('renders the logo', () => {
    const { getByAltText } = render(<Logo />)

    const logo = getByAltText('CÃ³ncavo')

    expect(logo).toHaveAttribute('src', '/logos/concavo_black.png')
  })
})
