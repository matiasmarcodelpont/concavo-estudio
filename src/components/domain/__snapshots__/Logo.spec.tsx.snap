// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo matches snapshot 1`] = `
<img
  alt="Logo CÃ³ncavo"
  height="66.66666666666667"
  src="/logos/concavo_black.png"
  width="200"
/>
`;
